src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageControl.xaml.cs

```diff
@@ -48,21 +48,20 @@ private void OnClearEnvironmentVariableGridError(object sender, EventArgs e)
             ClearGridError(dataGridEnvironmentVariables);
         }
 
-        private void OnFocusEnvironmentVariableGridRow(object sender, EventArgs e)
+        private async void OnFocusEnvironmentVariableGridRow(object sender, EventArgs e)
         {
             if (DataContext != null && DataContext is DebugPageViewModel)
             {
-                Dispatcher.BeginInvoke(new DispatcherOperationCallback((param) =>
+                await ThreadHelper.JoinableTaskFactory
+                    .WithPriority(VsTaskRunContext.UIThreadBackgroundPriority)
+                    .SwitchToMainThreadAsync();
```

Comment:
As a general rule, we like to see `SwitchToMainThreadAsync()` called on a regular JTF instance, and use that JTF instance to call `RunAsync` (or `StartOnIdle` as I describe above). That way, *any* call to `SwitchToMainThreadAsync` in this code or code it calls will all automatically inherit the priority you specify rather than just this one location.
I filed https://github.com/Microsoft/vs-threading/issues/140 to track considering a new analyzer to catch this.